web/backend/utils/calc.ts

```diff
@@ -126,21 +128,21 @@ export const calculateMetrics = async ({ mappedMetrics, log }: FastifyInstance):
                   const data = metricValue.value * 1000
                   if (data > 0) {
                     if (metricValue.labels?.quantile === 0.9) {
-                      serviceLatencyData.p90 = data
+                      serviceLatencyData.p90 += data
                       if (isEntrypointService) {
-                        aggregatedLatencyData.p90 = data
+                        aggregatedLatencyData.p90 += data
                       }
                     }
                     if (metricValue.labels?.quantile === 0.95) {
-                      serviceLatencyData.p95 = data
+                      serviceLatencyData.p95 += data
                       if (isEntrypointService) {
-                        aggregatedLatencyData.p95 = data
+                        aggregatedLatencyData.p95 += data
                       }
                     }
                     if (metricValue.labels?.quantile === 0.99) {
-                      serviceLatencyData.p99 = data
+                      serviceLatencyData.p99 += data
                       if (isEntrypointService) {
-                        aggregatedLatencyData.p99 = data
+                        aggregatedLatencyData.p99 += data
```

Comment:
You can't sum latencies. Some other math is required.

Reply:
5fdb2cf